### PR TITLE
Fix memory leak in cool modals

### PR DESCRIPTION
**Pull request by osdnk**
Fixes APP-1
Figma link (if any):

This was a regression caused by https://github.com/rainbow-me/rainbow/pull/1142

However, the original bug was already fixed in reanimated so this fix is not needed. Also, I expect that this particular line was unintentional anyway 

See the bug here (sorry for the image quality, this is the screenshot for the video):
before https://streamable.com/em0d2t
image

after: https://streamable.com/nvgoxe
image

Observe the memory chart changing: 

Before: https://streamable.com/e0kxpe
After:https://streamable.com/0vkpty

Before after 1 minute 
image

After fix after almost 2 minutes
image

(see recording for the documentation of the more precise experiments) 

There is still something broken (maybe this is just JS logic), but definitely, the derivative of the growth decreased so it's an improvement. 
 
## What changed (plus any additional context for devs)

I restored commented out line cause setting controller to `null`. This is important because in this structure (that is a bit incorrect, but ok) `controller` is holding `view` (`RNCMScreen` and `RNCMScreeView`) and vice versa. This is not java GC, but ARC (not having GC to be precise), which means that as long as the memory holds the reference for the object, it doesn't get deallocated. Here, one object keeps ref to another and the other way.

I fixed this in a way this was supposed to be fixed originally.

This was previously not discovered vie leak tool because of the fact that this is not an alienated object not being deallocated but the whole structure that was most likely too big to be noticed to be separated.

### CODE REVIEW

File: src/react-native-cool-modals/ios/RNCMScreen.m [hunk 1]
```diff
@@ -322,7 +322,7 @@ - (void)invalidate
 }
 
 - (void)removeController {
-  //_controller = nil;
+  _controller = nil;
```
Comment (osdnk) [inline src/react-native-cool-modals/ios/RNCMScreen.m:325]: this is the only meaningful change

**Review by terrysahaidak — APPROVED** [review]

**Review by tchayen — APPROVED** [review]

**Review by jkadamczyk — APPROVED** [review]

**Review by brunobar79 — CHANGES_REQUESTED** [review]
Seems like this is change is breaking e2e

**Comment by jkadamczyk** [comment]
@estrattonbailey Did you have any intentions with closing and reopening?
I will assign myself to that to see if we can merge it

**Comment by jkadamczyk** [comment]
It has some issues, we shouldn't merge it in this state. Gonna create linear tickets for that to address the memory leaks and link this PR

Uploading Screen Recording 2022-10-18 at 10.44.07.mov…

**Pull request closed without merging**